### PR TITLE
[doc] Fix info on the shape of pivots in `torch.lu` + more info on what and how they encode permutations.

### PR DESCRIPTION
**Pull request by nikitaved**
As per title.

### CODE REVIEW

File: torch/functional.py [hunk 1]
```diff
@@ -1449,7 +1449,12 @@ def _lu_impl(A, pivot=True, get_infos=False, out=None):
 
             - **factorization** (*Tensor*): the factorization of size :math:`(*, m, n)`
 
-            - **pivots** (*IntTensor*): the pivots of size :math:`(*, m)`
+            - **pivots** (*IntTensor*): the pivots of size :math:`(*, min(m, n))`.
+              ``pivots`` stores all the intermediate transpositions of rows.
+              The final permutation ``perm`` could be reconstructed by
+              applying :math:`swap(perm[i], perm[pivots[i] - 1])` for :math:`i = 0, \ldots, pivots.size(-1) - 1`,
```
Comment (vishwakftw) [inline torch/functional.py:1455]: `swap`, `perm` and `pivots` could be in `\text{}`, otherwise they'll be formatting like math.

Comment (nikitaved) [reply torch/functional.py]: replaced with `` `` to replicate the code insertion.

**Review by vishwakftw — APPROVED** [review]
Looks good to me, thanks.

**Pull request closed without merging**